### ISSUE TITLE
Change no-magic-numbers to support default argument values

### ISSUE THREAD

**Issue by LJNeon** [eslint/eslint#10751]
**What rule do you want to change?**
`no-magic-numbers`
**Does this change cause the rule to produce more or fewer warnings?**
`fewer`
**How will the change be implemented? (New option, new default behavior, etc.)?**
`new default behavior`, or `new option` if needed, but I don't believe it would be.
**Please provide some example code that this change will affect:**
```js
function format(ms, precision = 2) {
  // do stuff
}
```
**What does the rule currently do for this code?**
It considers the `2` a magic number, even though it has an explicit meaning.
**What will the rule do after it's changed?**
It would treat this code as valid.

**Comment by kaicataldo** [comment]
This seems like a reasonable enhancement to me :+1: I think this would have been added by default had this rule been written post ES2015.

**Comment by LJNeon** [comment]
Any progress on this change being made?

**Comment by platinumazure** [comment]
I've just added my :+1:. Per our [policy](https://eslint.org/docs/maintainer-guide/issues#accepting-issues), this needs one more :+1: from a team member and a champion from the team, then the proposal becomes accepted.

I've brought this up in our team chat-- hopefully we'll see some activity. I think this just fell through the cracks.

**Comment by ilyavolodin** [comment]
This issue now has enough up votes to be accepted. We still need a champion from the team, however. @eslint/eslint-team Anyone wants to champion this change?

**Comment by michaelficarra** [comment]
This is still a magic number. The program should be written like

```js
const IEEE_RECOMMENDED_PRECISION = 2;
const LJNEON_PREFERRED_PRECISION = 3;
const COMPANY_STANDARD_PRECISION = 4;

function format(ms, precision = LJNEON_PREFERRED_PRECISION) {
  // do stuff
}
```

to give context-free meaning to `2` instead of just the `format`-context sensitive meaning `precision`.

I continue to believe we should not add options that should be avoided in order to follow the either the spirit of or the initial justification for the rule.
